Modules/TPC/src/PIDClusterCheck.cxx

```diff
@@ -0,0 +1,95 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   PIDClusterCheck.cxx
+/// \author Laura Serksnyte
+///
+#include <iostream>
+
+#include "TPC/PIDClusterCheck.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+
+#include <fairlogger/Logger.h>
+// ROOT
+#include <TH1.h>
+#include <TList.h>
+#include <TPaveText.h>
+
+using namespace std;
+
+namespace o2::quality_control_modules::tpc
+{
+
+void PIDClusterCheck::configure(std::string) {}
+
+Quality PIDClusterCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  auto mo = moMap->begin()->second;
+  Quality result = Quality::Null;
+
+  // Check if the number of clusters is smaller than 159.
+  if (mo->getName() == "hNClusters") {
+    auto* h = dynamic_cast<TH1F*>(mo->getObject());
+    result = Quality::Good;
+    for (int i = 0; i < h->GetNbinsX(); i++) {
+      if (h->GetBinContent(i) > 0 && h->GetBinCenter(i) > 159) {
+        result = Quality::Bad;
+      }
+    }
+    // Call beautify function because the framework does not trigger it.
+    beautify(mo, result);
+  }
+
+  return result;
+}
+
+std::string PIDClusterCheck::getAcceptedType() { return "TH1"; }
+
+void PIDClusterCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality checkResult)
+{
+  std::cout << "BEAUTIFY FUNCTION OUTPUT! The Quality passed: " << checkResult << std::endl;
```

Comment:
You might want to use one of the logging interfaces if you would like to be sure that this message is seen in InfoLogger. That would be for example:
```
LOG(INFO) << "BEAUTIFY FUNCTION OUTPUT! The Quality passed: " << checkResult;
```
or:
```
#include "QualityControl/QcInfoLogger.h"
...
ILOG(Info) << "BEAUTIFY FUNCTION OUTPUT! The Quality passed: " << checkResult << ENDM
```